src/auth/Crypto.cc

```diff
@@ -348,24 +360,695 @@ int CryptoAES::validate_secret(const bufferptr& secret)
   return 0;
 }
 
-CryptoKeyHandler *CryptoAES::get_key_handler(const bufferptr& secret,
-					     string& error)
+CryptoKeyHandler *CryptoAES::get_key_handler(
+    const bufferptr& secret,
+		string& error)
 {
   CryptoAESKeyHandler *ckh = new CryptoAESKeyHandler;
   ostringstream oss;
-  if (ckh->init(secret, oss) < 0) {
+  if (ckh->init(secret, bufferptr(), oss) < 0) {
+    bufferptr a();
     error = oss.str();
     delete ckh;
     return NULL;
   }
   return ckh;
 }
 
+CryptoKeyHandler *CryptoAES::get_key_handler(
+    const bufferptr& secret,
+    const bufferptr& iv,
+    string& error)
+{
+  CryptoAESKeyHandler *ckh = new CryptoAESKeyHandler;
+    ostringstream oss;
+    if (ckh->init(secret, iv, oss) < 0) {
+      error = oss.str();
+      delete ckh;
+      return NULL;
+    }
+    return ckh;
+}
+
+
+class CryptoAES_256_CBC : public CryptoHandler {
+public:
+  CryptoAES_256_CBC() { }
+  ~CryptoAES_256_CBC() override {}
```

Comment:
please omit empty ctors/dtors (or `= default` if the compiler wouldn't otherwise generate them)